docs/reST/tut/newbieguide.rst

```diff
@@ -123,93 +127,61 @@ output file had the same pixel format as the input file.  If you're writing a
-misunderstanding the ``pygame.display.update()`` function.  With pygame, merely
-drawing something to the display surface doesn't cause it to appear on the
-screen -- you need to call ``pygame.display.update()``.  There are three ways
-of calling this function:
-
-
- * ``pygame.display.update()`` -- This updates the whole window (or the whole screen for fullscreen displays).
- * ``pygame.display.flip()`` -- This does the same thing, and will also do the right thing if you're using ``double-buffered`` hardware acceleration, which you're not, so on to...
- * ``pygame.display.update(a rectangle or some list of rectangles)`` -- This updates just the rectangular areas of the screen you specify.
-
-
-Most people new to graphics programming use the first option -- they update the
-whole screen every frame.  The problem is that this is unacceptably slow for
-most people.  Calling ``update()`` takes 35 milliseconds on my machine, which
-doesn't sound like much, until you realize that 1000 / 35 = 28 frames per
-second *maximum*. And that's with no game logic, no blits, no input, no AI,
-nothing.  I'm just sitting there updating the screen, and 28 fps is my maximum
-framerate. Ugh.
-
-The solution is called 'dirty rect animation'.  Instead of updating the whole
-screen every frame, only the parts that changed since the last frame are
-updated.  I do this by keeping track of those rectangles in a list, then
-calling ``update(the_dirty_rectangles)`` at the end of the frame.  In detail
-for a moving sprite, I:
-
- * Blit a piece of the background over the sprite's current location, erasing it.
- * Append the sprite's current location rectangle to a list called dirty_rects.
- * Move the sprite.
- * Draw the sprite at it's new location.
- * Append the sprite's new location to my dirty_rects list.
- * Call ``display.update(dirty_rects)``
-
-The difference in speed is astonishing. Consider that SolarWolf_ has dozens of
-constantly moving sprites updating smoothly, and still has enough time left
-over to display a parallax starfield in the background, and update that too.
-
-There are two cases where this technique just won't work. The first is where
-the whole window or screen really is being updated every frame -- think of a
-smooth-scrolling engine like an overhead real-time strategy game or a
-side-scroller.  So what do you do in this case?  Well, the short answer is --
-don't write this kind of game in pygame.  The long answer is to scroll in steps
-of several pixels at a time; don't try to make scrolling perfectly smooth.
-Your player will appreciate a game that scrolls quickly, and won't notice the
-background jumping along too much.
-
-A final note -- not every game requires high framerates. A strategic wargame
-could easily get by on just a few updates per second -- in this case, the added
-complexity of dirty rect animation may not be necessary.
-
-
-There is NO rule six.
----------------------
+Be wary of outdated, obsolete, and optional advice.
```

Comment:
Against removing this part. Especially since dirty rect optimization is still useful.

Reply:
Do you think it's useful to newbies though? Here are my main problems with the section:
- Keeping track of dirty rects is error prone and beyond the capabilities of most newbies, from what I've seen. I worry that telling them to try it will lead them down a path of frustration (e.g. "my game is running is slow, I should try using dirty rects to fix it" -> they struggle to implement it properly and the FPS doesn't even improve).
- Most beginner games can't benefit from dirty rect tracking anyways, due to having a moving camera + background image  (e.g. flappy bird, top-down survival, platformers). And simpler games that could benefit (like snake, puzzle games, pacman etc.) likely don't need it to achieve high performance in the first place.
- The way the section was written made it sound like pygame _can't_ run well without it, which makes pygame sound limited, slow & outdated. Prospective pygame users have asked about the display.update() -> 28 FPS maximum thing (this could be fixed with a rewrite)

I guess I'm just not sure who the target audience of this section would be, and what advice we'd want to convey beyond "this exists, but you probably don't need it?"

Reply:
Dirty rect optimisation is not that useful. What might be useful is not re-drawing the whole screen, and instead only erasing and re-drawing a small section, but specifically the dirty rects parameter to update seems like a micro-optimisation these days.